### DISCUSSION TITLE
Touch screen transformation

### DISCUSSION THREAD

**Discussion by julr** [KlipperScreen/KlipperScreen#515] [Ideas]
I am running KlipperScreen on a Raspberry Pi 4 with a 4.3 inch DSI touch display. For better access to the USB ports I want to rotate the screen by 180 degrees. This is easily done with the respective kernel command line parameters but the touch screen is not rotated. This makes it nearly impossible to use the screen. I quick and dirty added the call to `xinput` in the screen.py but proper support for touch screen transformation (e.g. via the config file) would be much appreciated. This is the call that I added:
```
    logging.info("KlipperScreen version: %s" % version)

    subprocess.Popen(["/usr/bin/xinput", "set-prop", "raspberrypi-ts", "Coordinate Transformation Matrix", "-1", "0", "1", "0", "-1", "1", "0", "0", "1"])

    win = KlipperScreen(args, version)
```

**Comment by alfrix** [comment]
> support for touch screen transformation (e.g. via the config file) would be much appreciated

klipperscreen is just an application, the screen and touchscreen itself are handled by the OS, nevertheless some examples are provided on how to accomplish this
https://klipperscreen.readthedocs.io/en/latest/Hardware/#touchscreen-touch-rotation
